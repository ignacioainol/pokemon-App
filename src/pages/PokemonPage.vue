<template>
  <h1 v-if="!pokemon">Waiting ...</h1>
  <div v-else>
    <h1>Quien es este Pokemon?</h1>

    <PokemonPicture 
        :pokemonId="pokemon.id" 
        :showPokemon="showPokemon"/>
    <PokemonOptions 
        :pokemons="pokemonArr"
        @selection="checkAswer"
        />
  </div>
</template>

<script>
import PokemonPicture from '@/components/PokemonPicture.vue'
import PokemonOptions from '@/components/PokemonOptions.vue'

import getPokemonOptions from '@/helpers/getPokemonOptions'

PokemonPicture
export default {

  components: {
    PokemonPicture,  
    PokemonOptions
  },
  data() {
    return{
      pokemonArr: [],
      pokemon: null,
      showPokemon: false
    }
  },
  methods: {
    async mixPokemonArray() {
      this.pokemonArr = await getPokemonOptions()
      const randInt = Math.floor(Math.random() * 4)
      this.pokemon = this.pokemonArr[randInt]
      console.log(this.pokemon) 
    },
    checkAswer( pokemonId){
      this.showPokemon = true
    }
  },
  mounted() {
    this.mixPokemonArray()
}
}
</script>