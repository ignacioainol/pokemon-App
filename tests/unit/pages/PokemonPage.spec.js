import PokemonPage from "@/pages/PokemonPage"
import { shallowMount, mount } from "@vue/test-utils"
import { pokemons } from "../mocks/pokemons.mock";


describe('PokemonPage Component Test', () => {

  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(PokemonPage)
  });

  test('Debe hacer match con el snapshot', () => {
    expect(wrapper.html()).z
  })

  test('debe llamar mixPokemonArray al montar', () => {
    const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')

    const wrapper = shallowMount(PokemonPage)

    expect(mixPokemonArraySpy).toHaveBeenCalled()
  })

  test('debe de hacer match con el snapshot cuando cargan los pokemons', () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: ''
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Debe de mostrar los componentes de PokemonPicture y PokemonOptions', () => {

    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemons,
          pokemon: pokemons[3],
          showPokemon: false,
          showAnswer: false,
          message: ''
        }
      }
    })

    const pokemonPicture = wrapper.find('pokemon-picture-stub')
    const pokemonOptions = wrapper.find('pokemon-options-stub')

    expect(pokemonPicture.exists()).toBeTruthy()
    expect(pokemonOptions.exists()).toBeTruthy()
    expect(pokemonPicture.attributes('pokemonid')).toBe('4')
    expect(pokemonOptions.attributes('pokemons')).toBeTruthy()
  })
})
