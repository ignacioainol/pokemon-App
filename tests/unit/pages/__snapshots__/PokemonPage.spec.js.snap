// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component Test Debe de mostrar los componentes de PokemonPicture y PokemonOptions 1`] = `
<div>
  <h1>Quien es este Pokemon?</h1>
  <pokemon-picture-stub pokemonid="4" showpokemon="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
</div>
`;

exports[`PokemonPage Component Test debe de hacer match con el snapshot cuando cargan los pokemons 1`] = `
<div>
  <h1>Quien es este Pokemon?</h1>
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
</div>
`;
